src/main/java/com/devicehive/service/AbstractHazelcastEntityService.java

```diff
@@ -61,15 +64,15 @@ protected void init() {
                               Date timestamp, String status,
                               Integer take, Boolean hasResponse,
                               HivePrincipal principal, Class<T> entityClass) {
-        final Predicate filters = hazelcastHelper.prepareFilters(devices, names, timestamp, status, hasResponse, principal);
+        final Predicate filters = hazelcastHelper.prepareFilters(getAvailableDevices(devices, principal), names, timestamp, status, hasResponse);
```

Comment:
@YermakovSergiy it's just a matter of coding style, and better to introduce new variable.